packages/vuetify/src/labs/VNumberInput/__tests__/VNumberInput.spec.cy.tsx

```diff
@@ -158,4 +158,97 @@ describe('VNumberInput', () => {
         .then(() => expect(numberInputValue.value).to.equal(0))
     })
   })
+
+  describe('parsing pasted values', () => {
+    it('should parse valid numbers correctly', () => {
+      const scenarios = [
+        { text: '-0', expected: '0' },
+        { text: '-00123', expected: '-123' },
+        { text: '.250', expected: '0.25' },
+        { text: '000.321', expected: '0.321' },
+        { text: '100.000', expected: '100' },
+      ]
+      scenarios.forEach(({ text, expected }) => {
+        cy.mount(() => <VNumberInput />)
+        cy.window().its('navigator.clipboard').then((clip: Clipboard) => clip.writeText(text))
+        cy.get('.v-number-input input').focus()
+        cy.document().invoke('execCommand', 'paste')
+        cy.get('.v-number-input input').blur().should('have.value', expected)
+      })
+    })
+
+    it('should retrieve digits when pasted value contains some mixed characters', () => {
+      const scenarios = [
+        { text: ' 1,250.32\n', expected: '1250.32' },
+        { text: 'text 1024 text', expected: '1024' },
+        { text: '- 1123.', expected: '-1123' },
+      ]
+      scenarios.forEach(({ text, expected }) => {
+        cy.mount(() => <VNumberInput />)
+        cy.window().its('navigator.clipboard').then((clip: Clipboard) => clip.writeText(text))
+        cy.get('.v-number-input input').focus()
+        cy.document().invoke('execCommand', 'paste')
+        cy.get('.v-number-input input').blur().should('have.value', expected)
+      })
+    })
+
+    it('should use as much characters as possible to display value containing only received digits', () => {
+      const scenarios = [
+        { text: '0.000010', expected: '0.00001' },
+        { text: '0.012340', expected: '0.01234' },
+        { text: '099999990.000000010', expected: '99999990.00000001' },
+        { text: '99999999999999999999999999999', expected: '999999999999999' },
```

Comment:
Why `99999999999999999999999999999` becomes `999999999999999` ?

Reply:
JS number has a quirk:

Precision loss from using numeric literals with more than 15 significant digits.

<img width="219" alt="Screenshot 2024-07-19 at 11 48 20 pm" src="https://github.com/user-attachments/assets/637c84b1-b301-424c-b895-68cbc8005f32">

Reply:
I think the end goal is the same. We want to respect whatever user typed and avoid replacing already typed characters because of JS quirks. Using integer limit/range with inputs that can be float values seems like something that will leave tons of scenarios uncovered.

Current implementation shortens input string (removing characters from the end) until condition is satisfied
```ts
cleanText !== Number(cleanText).toFixed(decimalDigits)
```

Original question was about a very long number. You can paste or type to any example from documentation. Pasted it becomes `1e+29`, typed... well breaks in more spectacular way and can lead to `NaN`, hence test case seems a great fit.

Reply:
> Using integer limit/range with inputs (that can be float values) seems like something that will leave tons of scenarios uncovered.

The primary purpose of number-input is **BASIC NUMERICAL INPUTS**, for example, money/weight etc. It should AVOID JS number quirks rather than RESOLVE all precision & unsafe JS quirks.

As per my understanding, there are three number challenges:

- Unsafe integer (`9007199254740993` -> `9007199254740992`) - use `MAX_SAFE_INTEGER` & `MIN_SAFE_INTEGER` range
- JS floating error (`0.1 + 0.2 === 0.30000000000000004 `) - almost impossible to resolve perfectly, so we make it clear in doc we use `toFixed` to cope with it
- Exponent format - not supported because they invalidate +/- buttons

These would cover the most of the use cases.

If users want to deal with crazy numbers (bigInt or long decimals), they should not use number-input but operate on STRING in text-field. These cases do not require +/- buttons anyway

Reply:
I always considered my implementation as a way to avoid JS quirks rather than resolving them :)

**Scenario**: For whatever reason user types (or pastes) `555555...` (very long string of digits). What would we like to show:
- `NaN` --- exposing JS quirk
- `9007199254740991` (MAX_SAFE_INTEGER) --- exposing JS quirk
- `5555555555555555` (digits from user, but only up until they can be displayed)

I am leaning towards the last and I still did not hear any argument against it. In my opinion it is not contrary to any of your points.

Imagine a developer that just placed `v-number-input` in the form and never expected end-user to have a "heavy-finger" or a slightly broken keyboard (I am used to both). But in some of the scenarios, he might unnecessarily surprise end-user. I think the unsurprising result is the best outcome and it it's implementation is quite unsophisticated.

I also don't want to fight over it. It is much more important to close this PR and proceed to strict precision implementation. If you feel like `Math.max(Math.min(Number(cleanText), Number.MAX_SAFE_INTEGER), Number.MIN_SAFE_INTEGER)` should replace the while-loop, I won't have anything against adjusting the PR.

Reply:
MAX_SAFE_INTEGER will be used as the max `prop.max`, so any input bigger than it will be restricted/changed to MAX_SAFE_INTEGER.